### ISSUE TITLE
"validation_not_unique" No Longer Working?

### ISSUE THREAD

**Issue by blakepatteson** [pocketbase/pocketbase#2868]
Not sure if this is by design or not, but I upgraded my pocketbase instances from 0.12.1 to 0.16.6 and no longer have a useful "validation_not_unique" message. 

When I create a unique index on a particular collection, at one point my PocketBase instance gave good validation messages when the record already exists in the collection, but it appears that this is no longer working (not sure if this is by design?)

See below : 

```
C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData'}'
<Response [400]>
{'code': 400, 'message': 'Failed to create record.', 'data': {}}

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData1'}'
<Response [200]>
{'collectionId': 'qdiex7lji2d7czd', 'collectionName': 'testData', 'created': '2023-07-06 18:03:48.195Z', 'id': 'l2nvlx1bo7ydbvk', 'testField': 'someTestData1', 'updated': '2023-07-06 18:03:48.195Z'}

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData'}'
<Response [400]>
{'code': 400, 'message': 'Failed to create record.', 'data': {}}

C:\zzz\repos\zpython>
```

I created a record with a collection "testData" with one field "testField" and gave it value "someTestData". Then when I tried to use the REST API to create the record again, it errored out with this message : 
```
{'code': 400, 'message': 'Failed to create record.', 'data': {}}
```
The 'data' property is empty, but previously the database was giving me good messages like this : 

`{"code":400,"message":"Something went wrong.","data":{"testField":{"code":"validation_not_unique","message":"Value must be unique."}}}`

Did this change when the "uniqueness" field updated with a previous PocketBase release (maybe with #544)? I liked having these error messages because I could easily understand why the record failed to be created. 

I saw some issues with similar (but not quite the same) problems : 
    #1766
    #1755
    #542
    #544
    #543
    #190
    
I was able to replicate it on a brand new deploy of a PocketBase instance on fly.io using this dockerfile : 

```
FROM alpine:latest

ARG PB_VERSION=0.16.6

RUN apk add --no-cache \
    unzip \
    # this is needed only if you want to use scp to copy later your pb_data locally
    openssh

# download and unzip PocketBase
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/

EXPOSE 8080

# start PocketBase
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```

Please let me know if any further documentation or something is needed to replicate, or if I'm just missing something obvious. Thanks!

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it on the live demo, aka. the correct unique index error is returned for me.

The general error could be returned in case the client request doesn't satisfy the Create API rule of your collection or the unique index is an expression.

If that's not the case, please export your collections configuration as JSON (Admin UI > Settings > Export collections) and I'll try to investigate it further.

**Comment by blakepatteson** [comment]
Thanks for getting back to me so quickly!!

I did try with all the "API Rules" unlocked, and still got the same issue :

```
C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData1'}'
<Response [200]>
{'collectionId': 'jd8pkq4jaqa9vd8', 'collectionName': 'testData', 'created': '2023-07-06 18:31:29.892Z', 'id': 'pxaexpaiabf8aq9', 'testField': 'someTestData1', 'updated': '2023-07-06 18:31:29.892Z'}

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData1'}'
<Response [400]>
{'code': 400, 'message': 'Failed to create record.', 'data': {}}

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'testField': 'someTestData1'}'
<Response [400]>
{'code': 400, 'message': 'Failed to create record.', 'data': {}}
```

Here's my "collections" export from "/_/#/settings/export-collections" : 
```
[
    {
        "id": "jd8pkq4jaqa9vd8",
        "name": "testData",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "sjabafzn",
                "name": "testField",
                "type": "text",
                "system": false,
                "required": false,
                "options": {
                    "min": null,
                    "max": null,
                    "pattern": ""
                }
            }
        ],
        "indexes": [
            "CREATE UNIQUE INDEX `idx_yw6wJwd` ON `testData` (`testField`)",
            "CREATE INDEX `idx_6gTfQ22` ON `testData` (`testField`)"
        ],
        "listRule": "",
        "viewRule": "",
        "createRule": "",
        "updateRule": "",
        "deleteRule": "",
        "options": {}
    }
]
```

So I think the client request should satisfy the "Create" API Rule of this collection - and I'm not sure if this "Unique Index" is an expression? 

And please excuse my ignorance, but if this "unique index is an expression", how should I create so it's not an expression? I looked through the docs for "unique" and couldn't find much about that particular issue.

**Comment by ganigeorgiev** [comment]
By "index expression" I meant column expressions - https://www.sqlite.org/expridx.html, this is not the case in your example.

It is not working because the SQLite error message is normalized to lowercase but the field name is left as it is when performing the validator checks and it fails to match. I'll update the tests and will push a fix in `master` a little bit later today.

_Also note that you don't need the second non-unique index._

**Comment by blakepatteson** [comment]
Cool - I'm glad you were able to figure that out so quick!

I tested with a lower case name and it worked as expected : 

```

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'test_field': 'someTestData1'}'
posting to 'https://[SOME-DOMAIN].fly.dev/api/collections/test_data/records'
<Response [200]>
{'collectionId': 'pc4mjrwon2qvgst', 'collectionName': 'test_data', 'created': '2023-07-06 19:08:28.343Z', 'id': 'lzvav2weqamn7ut', 'test_field': 'someTestData1', 'updated': '2023-07-06 19:08:28.343Z'}

C:\zzz\repos\zpython>python pocketbase_api.py
[PB_AUTH.SUCCESS] <Response [200]>
json_to_send : '{'test_field': 'someTestData1'}'
posting to 'https://[SOME-DOMAIN].fly.dev/api/collections/test_data/records'
<Response [400]>
{'code': 400, 'message': 'Failed to create record.', 'data': {'test_field': {'code': 'validation_not_unique', 'message': 'Value must be unique.'}}}
```

So I think you're spot on about the root cause.

And thanks for explaining the "index expression" and clarifying that I don't need that second index - I was just trying everything just in case!

**Comment by ganigeorgiev** [comment]
I've updated the tests and submitted a fix in `master`.

The fix will be available sometime later this week with v0.16.8.
